egs/wsj/s5/steps/nnet3/tdnn/make_configs.py

```diff
@@ -339,17 +355,25 @@ def MakeConfigs(config_dir, splice_indexes_string,
         prev_layer_output = nodes.AddAffRelNormLayer(config_lines, "Tdnn_{0}".format(i),
                                                     prev_layer_output, nonlin_output_dim, norm_target_rms = 1.0 if i < num_hidden_layers -1 else final_layer_normalize_target)
         # a final layer is added after each new layer as we are generating configs for layer-wise discriminative training
-        nodes.AddFinalLayer(config_lines, prev_layer_output, num_targets,
-                           use_presoftmax_prior_scale = use_presoftmax_prior_scale,
-                           prior_scale_file = prior_scale_file,
-                           include_log_softmax = include_log_softmax)
+
+        if not add_final_sigmoid:
```

Comment:
it feels strange to check for the not of a boolean. Could you switch the check to 
if add_final_sigmoid:

Also add a brief local comment about why this is done (i.e., for quadratic objectives)